ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -705,6 +705,51 @@ def evpn_gateway_remote_peers(self):
             return get(self._switch_data_combined, "evpn_gateway.remote_peers")
         return None
 
+    @cached_property
+    def mpls_vpn_gateway(self):
+        if self.underlay_router is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.enabled", default=False)
+
+    @cached_property
+    def evpn_domain_identifier(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.evpn_domain_id", default="65000:1")
+
+    @cached_property
+    def mpls_vpn_domain_identifier(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.mpls_domain_id", default="65000:2")
+
+    @cached_property
+    def mpls_vpn_domain_dpath_enabled(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.enable_d_path", default=True)
+
+    @cached_property
+    def mpls_vpn_gateway_address_families(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.address_families", default=["vpn-ipv4"])
+
+    @cached_property
+    def mpls_vpn_gateway_remote_peers(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.remote_peers", default=[])
+
+    @cached_property
+    def mpls_vpn_gateway_mpls_tunnel_source(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.mpls_tunnel_source", default="Loopback0")
+
+    @cached_property
+    def mpls_vpn_gateway_maximum_routes(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.maximum_routes", default=0)
+
+    @cached_property
+    def mpls_vpn_gateway_local_as(self):
+        if self.underlay_router is True and self.mpls_vpn_gateway is True:
+            return get(self._switch_data_combined, "mpls_vpn_gateway.local_as", default=False)
```

Comment:
We should not mix types for variables (default False) when expecting an AS
Better to do something like this which would return None if not present and be ignored in the template later:

```suggestion
    def mpls_vpn_gateway_local_as(self):
        if self.underlay_router is True and self.mpls_vpn_gateway is True:
            return get(self._switch_data_combined, "mpls_vpn_gateway.local_as")
```

Reply:
Implemented this in new "singe fact" dict for mpls_vpn_gateway.